### DISCUSSION TITLE
Differences between the two provided builds for Windows x64 in v9.0.0?

### DISCUSSION THREAD

**Discussion by hrxn** [sharkdp/fd#1456] [Q&A] [answered]
As before for **fd**, there's the following provided in the project's releases section:
```
fd-v9.0.0-x86_64-pc-windows-gnu.zip
fd-v9.0.0-x86_64-pc-windows-msvc.zip
```

While there's only the `windows-msvc` version for x86 - I'm aware that this is nothing new, has been like this since at least.. well, I've started to use *fd*.

If you use scoop, the `windows-msvc` build will be selected for both supported CPU architectures (https://github.com/ScoopInstaller/Main/blob/master/bucket/fd.json), and that's what I've been using as well, so far.

Without ever looking deeper into this, my assumption has been so far that `window-msvc` stands for, well MSVC, i.e. Microsoft's C and C++ runtime libaries, installed with the (in-)famous *Microsoft Visual C++ Redistributable* packages, which implies that these libraries are required for running this variant of `fd.` And `windows-gnu` targets the ususal GCC toolchain, I suppose. Without actually knowing what `rustc` does, but I guess this is not really relevant here. The only thing a user needs to know is that the GNU build does not require any further dependencies, I think.

My basic assumption held true thus far, as seen in many other different (CLI) programs written in Rust that also support Windows, which are provided in these same two variants as listed above too. And, unsurprisingly, the `windows-gnu` binary is usually a bit larger than the `windows-msvc` binary, which has been the case for basically all Rust-based projects I've used in the not too distant past, if I recall that correctly.

But this does not hold true for *fd* v9.0.0, where it's the other way around for the two binary sizes, and I wonder that means?
 
Just asking out of curiosity, basically. I mean, don't get me wrong, I don't really care if a program binary is ~ 2800 KiB or ~ 3100 KiB, even if I use it literally every day.

But since v9 is basically all about performance, I would be very interested in the differences in terms of performance characteristics, if there are actually any.

I've done a very simple ad-hoc test here for myself, basically just Powershell's `Measure-Command` of a very simple loop with 1000 repetitions running a simple *fd* search in a small (well, depends on whom you ask) directory structure with about 35k files, and from what I've observed here, the GNU binary seems about 2-3 seconds faster?

**Comment by sharkdp** [comment, marked as answer]
> Without ever looking deeper into this, my assumption has been so far that `window-msvc` stands for, well MSVC, i.e. Microsoft's C and C++ runtime libaries, installed with the (in-)famous _Microsoft Visual C++ Redistributable_ packages, which implies that these libraries are required for running this variant of `fd.` And `windows-gnu` targets the ususal GCC toolchain, I suppose. Without actually knowing what `rustc` does, but I guess this is not really relevant here. The only thing a user needs to know is that the GNU build does not require any further dependencies, I think.

That is my understanding as well.

> But this does not hold true for _fd_ v9.0.0, where it's the other way around for the two binary sizes, and I wonder that means?

Good question. I don't know. I could imagine that maybe dead code elimination / link time optimization worked better on GCC for some reason.

> But since v9 is basically all about performance, I would be very interested in the differences in terms of performance characteristics, if there are actually any.

That would be interesting indeed. I don't have access to a Windows machine, and therefore never made any benchmarks on Windows systems.
